BDArmory/Radar/RadarUtils.cs

```diff
@@ -2169,6 +2504,138 @@ public static bool TerrainCheck(Vector3 start, Vector3 end)
             return false;
         }
 
+        /// <summary>
+        /// Helper method: check if line intersects terrain OR water
+        /// </summary>
+        public static bool TerrainCheck(Vector3 start, Vector3 end, CelestialBody body)
+        {
+            if (!BDArmorySettings.IGNORE_TERRAIN_CHECK)
+            {
+                if (!BDArmorySettings.CHECK_WATER_TERRAIN)
+                    return Physics.Linecast(start, end, (int)LayerMasks.Scenery);
+
+                if (!Physics.Linecast(start, end, (int)LayerMasks.Scenery))
+                {
+                    float dummyR, dummyA;
+                    bool result = checkWater(start, end, body, -1f, out dummyR, out dummyA);
+                    return result;
+                }
+
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Helper method: check if line intersects terrain and gives range and angle of intersection. Note this check goes to up to sqrRange, though the boolean behavior is still restricted to between start and end
+        /// </summary>
+        public static bool TerrainCheck(Vector3 start, Vector3 end, CelestialBody body, float range, out float R, out float angle, bool ignoreSetting = false)
+        {
+            angle = 0f;
+            if (!BDArmorySettings.IGNORE_TERRAIN_CHECK)
+            {
+                float sqrDist = ((end.x - start.x) * (end.x - start.x) + (end.y - start.y) * (end.y - start.y) + (end.z - start.z) * (end.z - start.z));
```

Comment:
You could defer this calculation until it's actually needed and just use
```csharp
Vector3 offset = end - start;
if (Physics.Raycast(start, offset, out RaycastHit hitInfo, ...
...
return R * R < offset.sqrMagnitude;
```
The performance difference is going to be marginal, but it makes the code easier to read.